internal/provider/aws/provider.go

```diff
@@ -596,6 +596,12 @@ func (p *awsProvider) ReconcileFirewallRule(
 	}
 
 	if securityGroup == nil {
+		if !firewallRule.DeletionTimestamp.IsZero() {
+			// Object is in process of being deleted – likely going to disappear
+			log.Info("FirewallRule found with deletion timestamp and no security group created. Ignoring since object must be deleted")
+			return status, nil
+		}
+
```

Comment:
If I understand correctly, if we arrive here with a finalizer but without SecurityGroup, we will be stuck. 

From what I understand you don't have an easy way to update the firewallRule here, but it would be better to ensure in this case that the finalizer is emptied no ?